File: GanttChartLightLibraryDemos/Demos/Samples.Resources/WPF-CSharp/GanttChartDataGrid/Database/NumericDependencyTypeConverter.cs
using System;
using System.Net;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Documents;
using System.Windows.Ink;
using System.Windows.Input;
using System.Windows.Media;
using System.Windows.Media.Animation;
using System.Windows.Shapes;
using System.Windows.Data;
using System.Globalization;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using DlhSoft.Windows.Controls;
using System.ComponentModel;
using System.Linq;

namespace GanttChartDataGridDatabaseSample
{
    public class NumericDependencyTypeConverter : IValueConverter
    {
        // Converts a DependencyType value from/to an int value.
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            int intValue = (int)value;
            return (DependencyType)intValue;
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            DependencyType dependencyTypeValue = (DependencyType)value;
            return (int)dependencyTypeValue;
        }

        private static NumericDependencyTypeConverter instance;
        public static NumericDependencyTypeConverter Instance
        {
            get
            {
                if (instance == null)
                    instance = new NumericDependencyTypeConverter();
                return instance;
            }
        }
    }
}

File: GanttChartLightLibraryDemos/Demos/Samples.Resources/WPF-VisualBasic/GanttChartDataGrid/Database/NumericDependencyTypeConverter.vb
Imports System.Net
Imports System.Windows.Ink
Imports System.Windows.Media.Animation
Imports System.Globalization
Imports System.Collections.ObjectModel
Imports DlhSoft.Windows.Controls
Imports System.ComponentModel

Public Class NumericDependencyTypeConverter
    Implements IValueConverter

    ' Converts a DependencyType value from/to an int value.
    Public Function Convert(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.Convert
        Dim intValue As Integer = CInt(Fix(value))
        Return CType(intValue, DependencyType)
    End Function

    Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
        Dim dependencyTypeValue As DependencyType = CType(value, DependencyType)
        Return CInt(Fix(dependencyTypeValue))
    End Function

    'INSTANT VB NOTE: The variable instance was renamed since Visual Basic does not allow variables and other class members to have the same name:
    Private Shared instance_Renamed As NumericDependencyTypeConverter
    Public Shared ReadOnly Property Instance() As NumericDependencyTypeConverter
        Get
            If instance_Renamed Is Nothing Then
                instance_Renamed = New NumericDependencyTypeConverter()
            End If
            Return instance_Renamed
        End Get
    End Property
End Class
